apps/src/applab/applab.js

```diff
@@ -1897,27 +1897,43 @@ Applab.getIdDropdown = function (filterSelector) {
  * @private
  */
 Applab.getIdDropdownFromDom_ = function (documentRoot, filterSelector) {
-  var divApplabChildren = documentRoot.find('#divApplab').children();
-  var elements = divApplabChildren.toArray().concat(
-      divApplabChildren.children().toArray());
+  var elements = documentRoot.find('#designModeViz [id^="' + applabConstants.DESIGN_ELEMENT_ID_PREFIX + '"]');
 
   // Return all elements when no filter is given
   if (filterSelector) {
-    elements = $(elements).filter(filterSelector).get();
-  }
-
-  return elements
-      .sort(function (elementA, elementB) {
-        return elementA.id < elementB.id ? -1 : 1;
-      })
-      .map(function (element) {
-        return {
-          text: quote(element.id),
-          display: quote(element.id)
-        };
-      });
+    elements = elements.filter(filterSelector);
+  }
+
+  return elements.sort(byId).map(function (_, element) {
+    var id = quote(element.id.replace(new RegExp('^' + applabConstants.DESIGN_ELEMENT_ID_PREFIX), ''));
+    return {text: id, display: id};
+  }).get();
+};
+
+/**
+ * Returns a list of IDs currently present in the DOM of the current screen,
+ * including the screen, sorted by z-index.
+ */
+Applab.getIdDropdownForCurrentScreen = function () {
+  var screen = $('#designModeViz .screen').filter(function () {
+    return this.style.display !== 'none';
+  }).first();
+
+  var elements = screen.find('[id^="' + applabConstants.DESIGN_ELEMENT_ID_PREFIX + '"]').add(screen);
+
+  return elements.sort(byZIndex).map(function (_, element) {
+    return element.id.replace(new RegExp('^' + applabConstants.DESIGN_ELEMENT_ID_PREFIX), '');
+  }).get();
 };
 
+function byZIndex(a, b) {
```

Comment:
This looks like it will sort elements based on the html z-index attribute. I believe what the spec is asking for is to show the elements based on their ordering within design mode, which is based on the order of the children of the screen element. This sorting behavior could really use a test, too.